src/main/java/seedu/address/logic/parser/schedule/EditScheduleCommandParser.java

```diff
@@ -0,0 +1,78 @@
+package seedu.address.logic.parser.schedule;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static seedu.address.logic.parser.schedule.CliSyntax.PREFIX_CLIENT_INDEX;
+import static seedu.address.logic.parser.schedule.CliSyntax.PREFIX_SESSION_INDEX;
+import static seedu.address.logic.parser.schedule.CliSyntax.PREFIX_UPDATED_SESSION_INDEX;
+
+import java.util.stream.Stream;
+
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.commands.schedule.EditScheduleCommand;
+import seedu.address.logic.commands.schedule.EditScheduleCommand.EditScheduleDescriptor;
+import seedu.address.logic.parser.ArgumentMultimap;
+import seedu.address.logic.parser.ArgumentTokenizer;
+import seedu.address.logic.parser.Parser;
+import seedu.address.logic.parser.ParserUtil;
+import seedu.address.logic.parser.Prefix;
+import seedu.address.logic.parser.exceptions.ParseException;
+
+public class EditScheduleCommandParser implements Parser<EditScheduleCommand> {
+
+    /**
+     * Returns true if none of the prefixes contains empty {@code Optional} values in the given
+     * {@code ArgumentMultimap}.
+     */
+    private static boolean arePrefixesPresent(ArgumentMultimap argumentMultimap, Prefix... prefixes) {
+        return Stream.of(prefixes).allMatch(prefix -> argumentMultimap.getValue(prefix).isPresent());
+    }
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the EditScheduleCommand
+     * and returns an EditScheduleCommand object for execution.
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public EditScheduleCommand parse(String args) throws ParseException {
+        requireNonNull(args);
+        ArgumentMultimap argMultimap =
+                ArgumentTokenizer.tokenize(args, PREFIX_CLIENT_INDEX, PREFIX_SESSION_INDEX,
+                        PREFIX_UPDATED_SESSION_INDEX);
+
+        if (!arePrefixesPresent(argMultimap, PREFIX_CLIENT_INDEX, PREFIX_SESSION_INDEX, PREFIX_UPDATED_SESSION_INDEX)
+                || !argMultimap.getPreamble().isEmpty()) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT,
+                    EditScheduleCommand.MESSAGE_USAGE));
+        }
+
+        Index clientIndex = ParserUtil.parseIndex(argMultimap.getValue(PREFIX_CLIENT_INDEX).get());
+        Index sessionIndex = ParserUtil.parseIndex(argMultimap.getValue(PREFIX_SESSION_INDEX).get());
+        Index updateSessionIndex = ParserUtil.parseIndex(argMultimap.getValue(PREFIX_UPDATED_SESSION_INDEX).get());
```

Comment:
will this throw NoSuchElementException if the input does not include `su/...`? This issue arises when implementing payment/adding schedule remark features. I think would be better not to declare it here, since if the schedule remains unedited, it's already handled in line 71

Reply:
For now, it shows Invalid Command Format, I get what you mean, maybe after you implement those features that you can try them out? cuz now all 3 indexes are required before your features are being implemented